hmda/src/main/scala/hmda/persistence/filing/FilingState.scala

```diff
@@ -24,6 +25,14 @@ case class FilingState(filing: Filing = Filing(),
         } else {
           FilingState(this.filing, submission :: submissions)
         }
+      case SubmissionUpdated(updated) =>
+        if (submissions.map(_.id).contains(updated.id)) {
+          val updatedList = updated :: submissions.filterNot(s =>
```

Comment:
This seems like it could mess with the behavior of `GetLatestSubmission` in the `FilingPersistence` actor.  It would put the most recently updated Submission at the head of the list, which could cause problems if someone creates a new submission, and an old one finishes uploading/validating later.